src/primitives/hrp.rs

```diff
@@ -44,25 +44,22 @@ impl Hrp {
     /// [BIP-173]: <https://github.com/bitcoin/bips/blob/master/bip-0173.mediawiki>
     pub fn parse(hrp: &str) -> Result<Self, Error> { Ok(Self::parse_and_case(hrp)?.0) }
 
-    /// Parses the human-readable part (see [`Hrp::parse`] for full docs).
-    ///
-    /// # Panics
-    ///
-    /// If the string is invalid as defined by BIP-173.
+    /// Parses the human-readable part (see [`Hrp::parse`] for full docs). Replaces any invalid
+    /// characters with `X`.
```

Comment:
Since the `_unchecked` was meant to be named after stdlib functions that panic if input is incorrect perhaps we can justify the name with the following:
```
    /// Parses the human-readable part (see [`Hrp::parse`] for full docs).
    ///
    /// Does not check that `hrp` is valid according to BIP-173 but does check for valid ASCII
    /// values, replacing any invalid characters with `X`.
```

Reply:
Huh. Do you have an example of this? I don't think `_unchecked` functions normally panic. They usually just result in wrong behavior, or in the case of ones marked `unsafe`, cause UB. Panicking sounds like checking to me.

In any case, happy to change the docs to match your suggestion.

Reply:
ooph, should have looked this up before posting, I got two things wrong (what I had in mind is stdlib integers):
- u32 uses `unchecked_foo` not `foo_unchecked`
- they contain UB _not_ panic if input is wrong

> This results in undefined behavior if rhs is larger than or equal to the number of bits in self, i.e. when [checked_shr](https://doc.rust-lang.org/std/primitive.u32.html#method.checked_shr) would return None.

Reply:
Heh, searching the stdlib docs, it seems that everything *except* the arithmetic opcodes use `_unchecked`, while the arithimetic ones use `uncheckde_`.

But as for UB, I do think that's the most common case. I did consider doing literally no checks, which might've left us in that situation, but it seemed pretty easy to define the behavior in an obviously-wrong way so that we wouldn't need to stick `unsafe` onto the function.